packages/xo-web/src/xo-app/logs/log-alert-header.js

```diff
@@ -19,12 +24,32 @@ export default [
       }),
   })),
   provideState({
+    effects: {
+      restartFailedVms: () => async (
+        { failedVmsIds: vms },
+        { log: { jobId: id, scheduleId: schedule } }
+      ) => {
+        await runBackupNgJob({
+          id,
+          schedule,
+          vms,
+        })
+      },
+    },
     computed: {
       formattedLog: (_, { log }) => JSON.stringify(log, null, 2),
+      failedVmsIds: (_, { log }) =>
+        log === undefined || !isFailureTask(log) || isEmpty(log.tasks)
```

Comment:
Why `isEmpty(tasks)`?

Reply:
The failure can be in the job level.

Reply:
This is already handled by `isFailureTask(log)`, why do you need to test `log.tasks`?